doc/examples/contextual_enhancement.py

```diff
@@ -0,0 +1,273 @@
+from dipy.core.gradients import gradient_table
+
+# Read data
+fetch_stanford_hardi()
+img, gtab = read_stanford_hardi()
+data = img.get_data()
+
+# Add Rician noise
+np.random.seed = 1234
+data_noisy = add_noise(data, 3, 500, noise_type='rician')
+
+# Select a small part of it
+data_small = data[25:40, 65:80, 35:42]
+data_noisy_small = data_noisy[25:40, 65:80, 35:42]
+
+"""
+Fit an initial model to the data, in this case Constrained Spherical
+Deconvolution is used.
+"""
+
+# Perform CSD on the original data
+from dipy.reconst.csdeconv import auto_response
+from dipy.reconst.csdeconv import ConstrainedSphericalDeconvModel
+response, ratio = auto_response(gtab, data, roi_radius=10, fa_thr=0.7)
+csd_model_orig = ConstrainedSphericalDeconvModel(gtab, response)
+csd_fit_orig = csd_model_orig.fit(data_small)
+csd_shm_orig = csd_fit_orig.shm_coeff
+
+# Perform CSD on the original data + noise
+response, ratio = auto_response(gtab, data_noisy, roi_radius=10, fa_thr=0.7)
+csd_model_noisy = ConstrainedSphericalDeconvModel(gtab, response)
+csd_fit_noisy = csd_model_noisy.fit(data_noisy_small)
+csd_shm_noisy = csd_fit_noisy.shm_coeff
+
+"""
+Inspired by [RodriguesEurographics_], a lookup-table is created, containing
+rotated versions of the kernel :math:`P_t` sampled over a discrete set of
+orientations. In order to ensure rotationally invariant processing, the discrete
+orientations are required to be equally distributed over a sphere. Per default,
+a sphere with 100 directions is used.
+
+"""
+
+from dipy.denoise.enhancement_kernel import EnhancementKernel
+from dipy.denoise.shift_twist_convolution import convolve
+
+# Create lookup table
+D33 = 1.0
+D44 = 0.02
+t = 1
+k = EnhancementKernel(D33, D44, t)
+
+"""
+Visualize the kernel
+"""
+
+from dipy.viz import fvtk
+from dipy.data import get_sphere
+from dipy.reconst.shm import sf_to_sh, sh_to_sf
+ren = fvtk.ren()
+
+# convolve kernel with delta spike
+spike = np.zeros((7, 7, 7, k.get_orientations().shape[0]), dtype=np.float64)
+spike[3, 3, 3, 0] = 1
+spike_shm_conv = convolve(sf_to_sh(spike, k.get_sphere(), sh_order=8), k,
+                          sh_order=8, test_mode=True)
+
+sphere = get_sphere('symmetric724')
+spike_sf_conv = sh_to_sf(spike_shm_conv, sphere, sh_order=8)
+model_kernel = fvtk.sphere_funcs((spike_sf_conv * 6)[3,:,:,:],
+                                  sphere,
+                                  norm=False,
+                                  radial_scale=True)
+fvtk.add(ren, model_kernel)
+fvtk.camera(ren, pos=(30, 0, 0), focal=(0, 0, 0), viewup=(0, 0, 1), verbose=False)
+fvtk.record(ren, out_path='kernel.png', size=(900, 900))
+
+"""
+.. figure:: kernel.png
+   :align: center
+
+   Visualization of the contour enhancement kernel.
+"""
+
+"""
+Shift-twist convolution is applied on the noisy data
+"""
+
+# Perform convolution
+csd_shm_enh = convolve(csd_shm_noisy, k, sh_order=8)
+
+
+"""
+The Spherical Deconvolution Transform is applied to sharpen the ODF field.
```

Comment:
It's actually Sharpening Deoncvolution Transform, the original function docstring had a (now fixed) mistake.